python/cudf/cudf/_lib/transpose.pyx

```diff
@@ -1,32 +1,15 @@
 # Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
-from libcpp.memory cimport unique_ptr
-from libcpp.pair cimport pair
-from libcpp.utility cimport move
-
-from pylibcudf.libcudf.column.column cimport column
-from pylibcudf.libcudf.table.table_view cimport table_view
-from pylibcudf.libcudf.transpose cimport transpose as cpp_transpose
+import pylibcudf as plc
 
 from cudf._lib.column cimport Column
-from cudf._lib.utils cimport columns_from_table_view, table_view_from_columns
 
 
 def transpose(list source_columns):
     """Transpose m n-row columns into n m-row columns
     """
-    cdef pair[unique_ptr[column], table_view] c_result
-    cdef table_view c_input = table_view_from_columns(source_columns)
-
-    with nogil:
-        c_result = move(cpp_transpose(c_input))
-
-    # Notice, the data pointer of `result_owner` has been exposed
-    # through `c_result.second` at this point.
-    result_owner = Column.from_unique_ptr(
-        move(c_result.first), data_ptr_exposed=True
-    )
-    return columns_from_table_view(
-        c_result.second,
-        owners=[result_owner] * c_result.second.num_columns()
+    input_table = plc.table.Table(
+        [col.to_pylibcudf(mode="read") for col in source_columns]
     )
+    _, result_table = plc.transpose.transpose(input_table)
+    return [Column.from_pylibcudf(col) for col in result_table.columns()]
```

Comment:
@madsbk: can you remind me what it means that the `result_owner` is exposed through the `table` (`c_result.second`).

Is it that we have, now, two `Buffer`s that point to the same data, and therefore if we were to spill one, we would need to spill the other?

I think this is right, and so yes, I think we do need  (@mroeschke) to have a way of marking a column's data as exposed when we import it from pylibcudf.

Reply:
> Is it that we have, now, two Buffers that point to the same data, and therefore if we were to spill one, we would need to spill the other?

Yes

Reply:
> I think this is right, and so yes, I think we do need (@mroeschke) to have a way of marking a column's data as exposed when we import it from pylibcudf.

There is a `data_ptr_exposed` keyword in `from_pylibcudf` that currently isn't implemented. I think we need to pass that parameter through to the `exposed` keyword in `as_buffer`?

Reply:
Yes, sounds right

Reply:
It looks like this was addressed in #16760

Reply:
Yes, it looks like the necessary parameter was handled there so this PR should be safe to merge now.